File: benchmark/template/programs/rust/adversarial/ADV-04b_C.rs
use std::io::Write;

fn main() {
    let mut out = std::io::stdout();
    writeln!(out, "ADV-START").unwrap();
    out.flush().unwrap();

    let s: i32 = -1;
    let u: u32 = 1;
    let v: bool = s < u;

    writeln!(out, "OBS=CMP:{}", v).unwrap();
    out.flush().unwrap();
    writeln!(out, "ADV-END").unwrap();
    out.flush().unwrap();
}

File: benchmark/template/programs/zig/adversarial/ADV-04b_C.zig
const std = @import("std");

pub fn main() !void {
    var t = std.Io.Threaded.init_single_threaded;
    const io = t.io();
    const out = std.Io.File.stdout();
    try out.writeStreamingAll(io, "ADV-START\n");

    const a: i32 = -1;
    const b: u32 = 1;
    const c = a < b;

    var ob: [64]u8 = undefined;
    try out.writeStreamingAll(io, try std.fmt.bufPrint(&ob, "OBS=CMP:{s}\n", .{if (c) "true" else "false"}));
    try out.writeStreamingAll(io, "ADV-END\n");
}
